src/apps/relay/mainrelay.c

```diff
@@ -2779,1082 +2968,1086 @@ static void print_features(unsigned long mfn) {
-#endif /* defined(OPENSSL_THREADS) */
 
-static void adjust_key_file_name(char *fn, const char *file_title, int critical) {
-  char *full_path_to_file = NULL;
+static void openssl_load_certificates(void);
+/*
+ * openssl genrsa -out pkey 2048
+ * openssl req -new -key pkey -out cert.req
+ * openssl x509 -req -days 365 -in cert.req -signkey pkey -out cert
+ *
+ */
+static void openssl_setup(void) {
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "=========== Openssl =========\n");
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "OpenSSL compile-time version: %s (0x%lx)\n", OPENSSL_VERSION_TEXT,
+                OPENSSL_VERSION_NUMBER);
 
-  if (!fn[0]) {
-    TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "\nERROR: you must set the %s file parameter\n", file_title);
-    goto keyerr;
-  } else {
+#if !TLS_SUPPORTED
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "TLS is not supported\n");
+#elif TLSv1_3_SUPPORTED
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "TLS 1.3 supported\n");
+#elif TLSv1_2_SUPPORTED
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "TLS 1.2 supported\n");
+#elif TLSv1_1_SUPPORTED
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "TLS 1.1 supported\n");
+#elif TLSv1_SUPPORTED
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "TLS 1.0 supported\n");
+#endif
 
-    full_path_to_file = find_config_file(fn, 1);
-    {
-      FILE *f = full_path_to_file ? fopen(full_path_to_file, "r") : NULL;
-      if (!f) {
-        TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING, "cannot find %s file: %s (1)\n", file_title, fn);
-        goto keyerr;
-      } else {
-        fclose(f);
-      }
-    }
+#if !DTLS_SUPPORTED
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "DTLS is not supported\n");
+#elif DTLSv1_2_SUPPORTED
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "DTLS 1.2 supported\n");
+#elif DTLS_SUPPORTED
+  TURN_LOG_FUNC(TURN_LOG_LEVEL_INFO, "DTLS supported\n");
+#endif
 
-    if (!full_path_to_file) {
-      TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING, "cannot find %s file: %s (2)\n", file_title, fn);
-      goto keyerr;
-    }
+  THREAD_setup();
+  SSL_load_error_strings();
+  OpenSSL_add_ssl_algorithms();
 
-    strncpy(fn, full_path_to_file, sizeof(turn_params.cert_file) - 1);
-    fn[sizeof(turn_params.cert_file) - 1] = 0;
+#if !TLS_SUPPORTED
+  if (!turn_params.no_tls) {
+    TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "WARNING: TLS is not supported\n");
+    turn_params.no_tls = 1;
+  }
+#endif
 
-    free(full_path_to_file);
-    return;
+  if (!(turn_params.no_tls && turn_params.no_dtls) && !turn_params.cert_file[0]) {
+    TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING, "\nWARNING: certificate file is not specified, I cannot start TLS/DTLS "
+                                          "services.\nOnly 'plain' UDP/TCP listeners can be started.\n");
+    turn_params.no_tls = 1;
+    turn_params.no_dtls = 1;
   }
 
-keyerr : {
-  if (critical) {
+  if (!(turn_params.no_tls && turn_params.no_dtls) && !turn_params.pkey_file[0]) {
+    TURN_LOG_FUNC(TURN_LOG_LEVEL_WARNING, "\nWARNING: private key file is not specified, I cannot start TLS/DTLS "
```

Comment:
in my opinion, if the certificate file isn't specified, but dtls / tls is configured to be turned on, then the process should exit. There's no reason to allow a misconfiguration like this to continue.